server/controllers/auth/signup.js

```diff
@@ -1,14 +1,34 @@
-// modules 
+// modules
+const Joi = require('@hapi/joi');
 const bcrypt = require('bcryptjs');
 const uuid = require('uuid/v1');
 const jwt = require('jsonwebtoken');
 
 const database = require("../../database");
 
+const signupSchema = Joi.object({
+	email: Joi.string()
+		.email()
+    .min(4)
+    .max(50)
+    .required(),
+	password: Joi.string()
+		.min(6)
+    .max(72)
+    .required(),
+});
+
 exports.signup = (req, res, next) => {
-  const email = req.body.email;
-  // password cannot be longer than 72 characters
-  const password = req.body.password;
+
+	const valid = signupSchema.validate(req.body);
+
+	if (valid.error) {
+		res.status(422);
+		next(valid.error);
```

Comment:
Why do we need `next()` after sending the response?

Reply:
`res.status(422);` is to set the status. `next` actually handles the error response via middlewares.

Reply:
but I think it is giving some kind of error https://github.com/logchimp/logchimp/pull/12#issuecomment-581121068

Reply:
No no, that error is because there is next after response dispatch. `res.status(422);` doesn't dispatch response. It only updates the response status.